snuba/consumers/consumer_builder.py

```diff
@@ -261,7 +272,9 @@ def build_streaming_strategy_factory(
                 table_writer,
                 metrics=self.metrics,
                 replacements_producer=(
-                    self.producer if self.replacements_topic is not None else None
+                    self.replacements_producer
+                    if self.replacements_topic is not None
+                    else None
```

Comment:
This could just be `self.replacements_producer`. It's always defined and should already be set to None in cases where there's no replacement topic.

Reply:
Yep, this is true